packages/swap/src/quoting/QuoteRequest.ts

```diff
@@ -0,0 +1,487 @@
+    if (!usdValue) {
+      logger.error(
+        `could not get usd value for DCA quote calculation. asset: ${this.srcAsset} , amount: ${this.depositAmount}`,
+      );
+      return;
+    }
+    if (Number(usdValue) <= usdChunkSize) {
+      return;
+    }
+    const numberOfChunks = Math.ceil(Number(usdValue) / usdChunkSize);
+
+    if (numberOfChunks > MAX_NUMBER_OF_CHUNKS) {
+      logger.info('number of chunks is bigger than max', { numberOfChunks, MAX_NUMBER_OF_CHUNKS });
+      return;
+    }
+
+    this.dcaQuoteParams = {
+      chunkSize: BigInt(
+        new BigNumber(this.depositAmount.toString()).dividedBy(numberOfChunks).toFixed(0),
+      ),
+      numberOfChunks,
+      additionalSwapDurationSeconds: Math.ceil(
+        env.DCA_CHUNK_INTERVAL_BLOCKS *
+          CHAINFLIP_STATECHAIN_BLOCK_TIME_SECONDS *
+          (numberOfChunks - 1), // we deduct 1 chunk because the first one is already accounted for in the regular quote
+      ),
+      chunkIntervalBlocks: env.DCA_CHUNK_INTERVAL_BLOCKS,
+    };
+  }
+
+  private async setLimitOrders() {
+    if (this.limitOrders) return this.limitOrders;
+    this.limitOrders = await this.quoter.getLimitOrders(
+      this.srcAsset,
+      this.destAsset,
+      this.depositAmount,
+    );
+    return this.limitOrders;
+  }
+
+  private async setBoostQuoteParams() {
+    const boostDepositsEnabled = await getBoostSafeMode(this.srcAsset).catch(() => true);
+    if (!boostDepositsEnabled) return;
+
+    const { estimatedBoostFeeBps, maxBoostFeeBps } = await getBoostFeeBpsForAmount({
+      amount: this.depositAmount,
+      asset: this.srcAsset,
+    });
+    this.estimatedBoostFeeBps = estimatedBoostFeeBps;
+    this.maxBoostFeeBps = maxBoostFeeBps;
+  }
+
+  private async setIndexPrices() {
+    [this.srcAssetIndexPrice, this.destAssetIndexPrice] = await Promise.all([
+      getAssetPrice(this.srcAsset).catch(() => undefined),
+      getAssetPrice(this.destAsset).catch(() => undefined),
+    ]);
+  }
+
+  private async getPoolQuote(): Promise<RegularQuote>;
+  private async getPoolQuote(opts: { boostFeeBps: number }): Promise<RegularBoostQuote>;
+  private async getPoolQuote(opts: { dcaParams: DcaParams }): Promise<DCAQuote>;
+  private async getPoolQuote(opts: {
+    boostFeeBps: number;
+    dcaParams: DcaParams;
+  }): Promise<DCABoostQuote>;
+  private async getPoolQuote({
+    boostFeeBps,
+    dcaParams,
+  }: { boostFeeBps?: number; dcaParams?: DcaParams } = {}) {
+    const includedFees = [];
+    const excludeFees: SwapFeeType[] = [];
+    let cfRateInputAmount = this.depositAmount;
+
+    // After this ticket, boost fee should be included in the response so dont have to calculate it ourselves
+    // https://linear.app/chainflip/issue/PRO-1370/include-boost-fees-in-quote-from-cf-swap-rate-v2
+    if (boostFeeBps) {
+      const boostFee = getPipAmountFromAmount(this.depositAmount, boostFeeBps);
```

Comment:
would be great if we can remove this condition

Reply:
nice let me see